### PR TITLE
Passing tests! Fixes #4

### LINKED ISSUE #4: codelab.html tests are currently failing
This is due to a neon-animated-pages depe (iron-selectable-behavior) that updated in the last release to use Polymer's new `observerNodes` API instead of a MutationObserver. This shouldn't have broken us, but the commit removed the `items` getter, which gave a live query of the distributed nodes. We were using `this.$.pages` in google-codelab to set the steps. Unfortunately, the upstream removal + change means neon-animated-pages doesn't have a populated `.items` when google-codelab is attached to the done. The easiest solution is to introduce a new event `google-codelab-ready` that fires when neon-animated-pages gets it's `.items`.

https://github.com/PolymerElements/iron-selector/commit/88658f25203084f578c502a15a89c0ee1ff6fd26

### PR DESCRIPTION
**Pull request by ebidel**
R: @crhym3 @marcacohen

### CODE REVIEW

File: codelab-style.html [hunk 1]
```diff
@@ -17,7 +17,7 @@
 <link rel="import" href="../paper-styles/color.html">
 
 <dom-module id="codelab-style">
-  <template>
+  <template strip-whitespace>
```
Comment (ebidel) [inline codelab-style.html]: oh. This strips whitespace from templates, which turns out, is a micro optimization in Blink's style recalcs for white space text nodes :\

Comment (ebidel) [inline codelab-style.html]: The timing issues in the tests spawn from change in https://github.com/PolymerElements/iron-selector/commit/92b0bbc31ef7dbb9bbbd1beb0db75877e3babcca

File: google-codelab.html [hunk 2]
```diff
@@ -459,6 +466,9 @@ <h1 class="title">{{title}}</h1>
       },
 
       _isLastStep: function(stepIndex) {
+        if (!this.steps.length) {
+          return false;
+        }
         return !this.steps.length || stepIndex >= this.steps.length - 1;
```
Comment (crhym3) [inline google-codelab.html]: I think one of `!this.steps.length` is redundant at this point.

File: codelab-style.html (hunk 1 above)
Comment (crhym3) [inline codelab-style.html]: Good find!

**Comment by crhym3** [comment]
Reviewed.

File: google-codelab.html (hunk 2 above)
Comment (ebidel) [inline google-codelab.html]: Done

**Comment by ebidel** [comment]
Going to merge and release. Safari/FF are currently broken b/c of the iron-selector changes.

**Pull request merged**